### PR TITLE
Pointer analysis tidy

### PR DESCRIPTION
**Pull request by martin-cs**
Profiling found these unnecessary loops in the pointer analysis.  They have a big impact on larger code bases.

@danpoe This is another branch you'll need for the next delivery.

### CODE REVIEW

File: src/pointer-analysis/value_set_analysis_fi.cpp [hunk 1]
```diff
@@ -212,21 +213,16 @@ void value_set_analysis_fit::add_vars(
     std::set<irep_idt> locals;
     get_local_identifiers(f_it->second, locals);
 
-    forall_goto_program_instructions(i_it, f_it->second.body)
+    for(std::set<irep_idt>::const_iterator
```
Comment (peterschrammel) [inline src/pointer-analysis/value_set_analysis_fi.cpp]: ranged for?

File: src/pointer-analysis/value_set_analysis_fi.cpp [hunk 2]
```diff
@@ -269,15 +265,16 @@ bool value_set_analysis_fit::check_type(const typet &type)
...
+    {
       case TRACK_ALL_POINTERS:
         { return true; break; }
       case TRACK_FUNCTION_POINTERS:
       {
         if(type.id()==ID_pointer)
         {
           const typet *t = &type;
-          while (t->id()==ID_pointer) t = &(t->subtype());
+          while(t->id()==ID_pointer) t = &(t->subtype());
```
Comment (peterschrammel) [inline src/pointer-analysis/value_set_analysis_fi.cpp]: cpplint

File: src/pointer-analysis/value_set_analysis_fivr.cpp [hunk 3]
```diff
@@ -84,33 +84,26 @@ void value_set_analysis_fivrt::add_vars(
   entry_cachet entry_cache;
 
   value_set_fivrt &v=state.value_set;
+  v.add_vars(globals);
 
-  for(goto_programt::instructionst::const_iterator
-      i_it=goto_program.instructions.begin();
-      i_it!=goto_program.instructions.end();
-      i_it++)
+  for(goto_programt::decl_identifierst::const_iterator
```
Comment (peterschrammel) [inline src/pointer-analysis/value_set_analysis_fivr.cpp]: ranged for?

File: src/pointer-analysis/value_set_analysis_fivr.cpp [hunk 4]
```diff
@@ -212,21 +206,16 @@ void value_set_analysis_fivrt::add_vars(
     std::set<irep_idt> locals;
     get_local_identifiers(f_it->second, locals);
 
-    forall_goto_program_instructions(i_it, f_it->second.body)
+    for(std::set<irep_idt>::const_iterator
```
Comment (peterschrammel) [inline src/pointer-analysis/value_set_analysis_fivr.cpp]: ranged for

File: src/pointer-analysis/value_set_analysis_fivr.cpp [hunk 5]
```diff
@@ -269,15 +258,16 @@ bool value_set_analysis_fivrt::check_type(const typet &type)
...
+    {
       case TRACK_ALL_POINTERS:
         { return true; break; }
       case TRACK_FUNCTION_POINTERS:
       {
         if(type.id()==ID_pointer)
         {
           const typet *t = &type;
-          while (t->id()==ID_pointer) t = &(t->subtype());
+          while(t->id()==ID_pointer) t = &(t->subtype());
```
Comment (peterschrammel) [inline src/pointer-analysis/value_set_analysis_fivr.cpp]: cpplint

File: src/pointer-analysis/value_set_analysis_fivrns.cpp [hunk 6]
```diff
@@ -84,33 +84,26 @@ void value_set_analysis_fivrnst::add_vars(
   entry_cachet entry_cache;
 
   value_set_fivrnst &v=state.value_set;
+  v.add_vars(globals);
 
-  for(goto_programt::instructionst::const_iterator
-      i_it=goto_program.instructions.begin();
-      i_it!=goto_program.instructions.end();
-      i_it++)
+  for(goto_programt::decl_identifierst::const_iterator
```
Comment (peterschrammel) [inline src/pointer-analysis/value_set_analysis_fivrns.cpp]: ranged for

File: src/pointer-analysis/value_set_analysis_fivrns.cpp [hunk 7]
```diff
@@ -212,21 +206,16 @@ void value_set_analysis_fivrnst::add_vars(
     std::set<irep_idt> locals;
     get_local_identifiers(f_it->second, locals);
 
-    forall_goto_program_instructions(i_it, f_it->second.body)
+    for(std::set<irep_idt>::const_iterator
```
Comment (peterschrammel) [inline src/pointer-analysis/value_set_analysis_fivrns.cpp]: ranged for

File: src/pointer-analysis/value_set_analysis_fivrns.cpp [hunk 8]
```diff
@@ -269,15 +258,16 @@ bool value_set_analysis_fivrnst::check_type(const typet &type)
...
+      {
         case TRACK_ALL_POINTERS:
           { return true; break; }
         case TRACK_FUNCTION_POINTERS:
         {
           if(type.id()==ID_pointer)
           {
             const typet *t = &type;
-            while (t->id()==ID_pointer) t = &(t->subtype());
+            while(t->id()==ID_pointer) t = &(t->subtype());
```
Comment (peterschrammel) [inline src/pointer-analysis/value_set_analysis_fivrns.cpp]: cpplint

**Comment by martin-cs** [comment]
I have rebased this (which fixes a number of linting issues) and added ranged for's where suggested.  The AppVeyor issues don't appear to be related to this change.

**Comment by peterschrammel** [comment]
@martin-cs, you seem to have done a pull instead of a pull --rebase.

**Comment by martin-cs** [comment]
I thought I rebased but two of the patches in the set were obsoleted by
fixes in the linting.  I don't know if rebase did something "smart" with
these.

**Comment by martin-cs** [comment]
No; what I did was much much worse.  But it is fixed now.

**Pull request merged**